modules/nf-core/limma/differential/templates/limma_de.R

```diff
@@ -317,6 +318,19 @@ if (!is.null(opt\$use_voom) && opt\$use_voom) {
 
     # Run voom to transform the data
     voom_result <- voom(dge, design)
+
+    if (!is.null(opt\$mixed_model) && opt\$mixed_model) {
+
+        corfit_args <- list(object = voom_result, design = design, block = sample.sheet[[opt\$block]])
+        corfit = do.call(duplicateCorrelation, corfit_args)
+
+        voom_args <- list(counts = dge, design = design, plot = FALSE, correlation = corfit\$consensus.correlation)
+        voom_result <- do.call(voom, voom_args)
+
+        corfit_args <- list(object = voom_result, design = design, block =  sample.sheet[[opt\$block]])
+        corfit = do.call(duplicateCorrelation, corfit_args)
+    }
```

Comment:
```suggestion
}

if (!is.null(opt\$block) {
    corfit = duplicateCorrelation, voom_result, design = design, block = sample.sheet[[opt\$block]])

    if (!is.null(opt\$use_voom) && opt\$use_voom) {
        data_for_fit <- voom(counts = dge, design = design, plot = FALSE, correlation = corfit\$consensus.correlation)
    }
}

if (!is.null(opt\$use_voom) && opt\$use_voom) {
```

Use of `block` and `duplicateCorrelation` is not voom-specific, so we should do something like this. It will require you to split the Voom conditional slightly. So the line above will also have to be changed from:

```
voom_result <- voom(dge, design)
```

to

```
data_for_fit <- voom(dge, design)
```